documentation/styleguide-doc.md

```diff
@@ -0,0 +1,167 @@
+# Documentation Coding Style
+
+## Where are the docs?
+
+TL;DR version: [`docs`](https://github.com/StackExchange/dnscontrol/tree/master/docs) is the [marketing website](https://dnscontrol.org). [`documentation`](https://github.com/StackExchange/dnscontrol/tree/master/documentation) is the [docs.dnscontrol.org](https://docs.dnscontrol.org/) website. (Yes, the names are backwards!)
+
+**The two websites**
+
+1. <https://dnscontrol.org/>
+   * The main website
+   * Source code: [`docs`](https://github.com/StackExchange/dnscontrol/tree/master/docs)
+   * Mostly "marketing" for the project.
+   * Rarely changes.  Updated via GitHub "pages" feature.
+2. <https://docs.dnscontrol.org/>
+   * Project documentation
+   * Source code: [`documentation`](https://github.com/StackExchange/dnscontrol/tree/master/documentation)
+   * Users and developer documentation
+   * Changes frequently.  Updated via [GitBook](https://www.gitbook.com/)
+
+**The directory structure**
+
+Within the git repo, docs are grouped:
+
+* [`documentation/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation): general docs
+* [`documentation/providers/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/providers/): One file per provider
+* [`documentation/functions/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/functions/): One file per `dnsconfig.js` language feature
+* [`documentation/assets/gcloud/`](https://github.com/StackExchange/dnscontrol/tree/master/documentation/assets/gcloud): Images for page Google Cloud DNS (PNGs only, please!)
+
+## How to add a new page?
+
+1. Add the page to the `documentation` (or a sub folder)
+2. List the page in `SUMMARY.md` so that it will appear in the table of contents, sidebar, etc.
+
+## Documentation previews
+
+> "Preview links are only accessible by GitBook users. We're working on a feature that will allow preview links to be viewable by anyone who accesses the PR." — _[GitBook](https://docs.gitbook.com/product-tour/git-sync/github-pull-request-preview#how-to-access-preview-links)_
+
+## Formatting tips
+
+### General
+
+Break lines every 80 chars.
+
+Include a blank line between paragraphs.
+
+Leave one blank line before and after a heading.
+
+Javascript code should use double quotes (`"`) for strings, not single quotes
+(`'`).  They are equivalent but consistency is good.
+
+### Headings
+
+```markdown
+#  Title of the page
+
+## Heading
+
+At least one paragraph.
+
+## Subheadings
+
+At least one paragraph.
+
+* **Step 1: Foo**
+
+Description of the step.
+
+* **Step 2: Bar**
+
+Description of the step.
+
+(further sub sub headings are discouraged encouraged)
+```
+
+### Code Snippets
+
+> "Currently, we don't support adding code blocks inside other code blocks. We don't have plans to support that functionality at this time." — _GitBook Support (2023-03-29)_
```

Comment:
The goal is to give the user something they can easily understand for including code with and without a filename.  Shorter examples are probably better (just 2-3 lines of code).  Maybe use a PNG to show what it looks like but link to something they can copy and paste?

Reply:
I agree with you! I think removing the GitBook notes is enough. There are two examples, each with a deep link to the GitHub source.

Reply:
Fixed by https://github.com/StackExchange/dnscontrol/commit/9a65c77a2d1d1925d927aab45341bf5266dbd0c7. @tlimoncelli would you like to close this conversation after reviewing it?

Reply:
_<https://docs.dnscontrol.org/~/revisions/BRtPCFXEcJg89ztpW8cK/developer-info/styleguide-doc#code-snippets>_